e3sm_diags/driver/aerosol_budget_driver.py

```diff
@@ -137,79 +70,206 @@ def run_diag(parameter: CoreParameter) -> CoreParameter:
+        The climo season.
+
+    Returns
+    -------
+    xr.DataArray
+        The column integrated mass.
+    """
+    if aerosol == "ncl":
+        aerosol_name = "SEASALT"
+    elif aerosol == "dst":
+        aerosol_name = "DUST"
+    else:
+        aerosol_name = aerosol.upper()
+
+    try:
+        # if ABURDEN terms are available, use them
+        var_key = f"ABURDEN{aerosol_name}"
+        burden = test_ds.get_climo_dataset(var_key, season)[var_key]
+    except IOError:
+        # if not, use the Mass_ terms and integrate over the column
+        var_key = f"Mass_{aerosol}"
+        ds_mass = test_ds.get_climo_dataset(var_key, season)
+        mass = ds_mass[var_key]
+
+        if mass.attrs["units"] != "kg/kg":
+            raise RuntimeError(
+                f"ERROR in aerosol_budget_driver/calc_column_integral!"
+                f"Mass_{aerosol} at season {season} has units {mass.units}."
+                f"But kg/kg units were expected."
             )
-            for key, values in metrics_dict_test.items():
-                writer.writerow([SPECIES_NAMES[key]])
-                for value in values:
-                    line = []
-                    line.append(value)
-                    line.append(values[value])
-                    line.append(metrics_dict_ref[key][value])
-                    writer.writerows([line])
-                writer.writerows([""])
 
-    logger.info(f"Metrics saved in {fnm}")
+        pressure_levs = _hybrid_to_pressure(
+            ds_mass, var_key, p0=100000.0, a_key="hyai", b_key="hybi"
+        )
+        # Preserve the original units.
+        pressure_levs.attrs["units"] = ds_mass[var_key].attrs["units"]
 
-    return parameter
+        # (72,lat,lon)
+        plev_key = xc.get_dim_keys(pressure_levs, axis="Z")
+        delta_p = pressure_levs.diff(plev_key)
+
+        # Perform numpy index-based arithmetic instead of xarray label-based
+        # arithmetic because the Z dims of `mass` and `delta_p` can have
+        # different names ("lev" vs. "ilev") with slightly different values. The
+        # CDAT version of this code uses numpy index-based arithmetic too.
+        with xr.set_options(keep_attrs=True):
+            # mass density * mass air   kg/m2
+            mass_3d = mass.copy()
+            mass_3d.values = mass.values * delta_p.values / 9.8
+
+        # kg/m2
+        lev_key = xc.get_dim_keys(mass_3d, axis="Z")
+        burden = mass_3d.sum(dim=lev_key, keep_attrs=True)
+
+    return burden
+
+
+def global_integral(var: xr.DataArray, area_m2: xr.DataArray) -> float:
+    """Compute global integral of 2 dimensional properties
+
+    Parameters
+    ----------
+    var : xr.DataArray
+        The variable.
+    area_m2 : xr.DataArray
+        The area in m2 units.
+
+    Returns
+    -------
+    float
+        The global integral of 2 dimensional properities
+    """
+    # TODO: Can use Xarray's function
+    # TODO: Replace axis=0 with the label name.
```

Comment:
I think it is fine to use `np.sum()` like before, if we always expect `axis=0` to be the Z axis.